mypy/plugins/dataclasses.py

```diff
@@ -200,6 +201,16 @@ def transform(self) -> None:
 
         self.reset_init_only_vars(info, attributes)
 
+        if (decorator_arguments['match_args'] and
+                ('__match_args__' not in info.names or
+                 info.names['__match_args__'].plugin_generated) and
+                attributes):
+            str_type = ctx.api.named_type("builtins.str")
+            literals: List[Type] = [LiteralType(attr.name, str_type)
+                        for attr in attributes if attr.is_in_init]
+            match_args_type = TupleType(literals, ctx.api.named_type("builtins.tuple"))
+            add_attribute_to_class(ctx.api, ctx.cls, "__match_args__", match_args_type, final=True)
```

Comment:
```suggestion
            # TODO this should have final=True, but that breaks multiple inheritance
            # https://github.com/python/mypy/pull/10191#issuecomment-1013399068
            add_attribute_to_class(ctx.api, ctx.cls, "__match_args__", match_args_type)
```

Reply:
We should probably special-case `__match_args__` to be ignored from `is_compatible_with_super` check 🤔